packages/node_modules/pouchdb-adapter-indexeddb/src/rewrite.js

```diff
@@ -101,7 +101,12 @@ function rewrite(data) {
     } else if (typeof data[key] === 'boolean') {
       clone[safeKey] = data[key] ? IDB_TRUE : IDB_FALSE;
     } else if (typeof data[key] === 'object') {
-      clone[safeKey] = rewrite(data[key]);
+      const val = rewrite(data[key]);
+      if (val) {
+        clone[safeKey] = val;
+      } else {
+        clone[safeKey] = data[key];
+      }
```

Comment:
This code looks surprising.  Would it also work to change the top of the function from

```js
function rewrite(data) {
  if (!needsRewrite(data)) {
    return false;
  } 
  ...
```
to e.g.

```js
function rewrite(data) {
  if (!needsRewrite(data)) {
    return data;
  } 
  ...
```

?

Reply:
That was the first fix I had. But this change seems more defensive because in

https://github.com/pouchdb/pouchdb/blob/c2d05ade1d23ec06328d8e68328de031bc8f456e/packages/node_modules/pouchdb-adapter-indexeddb/src/bulkDocs.js#L200-L206

the code seems to branch depending on whether rewriting happened or not.

Reply:
Good spot.  Any idea why calling `delete doc.data._attachments;` depends on whether rewriting happens?!

Reply:
I have no idea. All of these troubles seem to come from this commit:

https://github.com/pouchdb/pouchdb/commit/0cd8304e68693a6c47656c98e8aed188ebdb07f5#diff-5869c15c41cf71179c16a81e421d78dafce21dfe37bc47707513bc2809b49c53

Which introduces those difference branches without explaining it. From my limited understanding of this adapter the data attribute is only used for indexeddb views and those views will most likely not operate on attachments. So I guess we could also drop them always, but I really don’t know.

Reply:
> I have no idea. All of these troubles seem to come from this commit:

It looks like `_attachments` may have already been rewritten to `_c95_attachments` at that point.  If that's the case, `delete ...._attachments` would be a no-op.  I'll see if that can be cleaned up a bit.

Reply:
Let's see if this passes: https://github.com/pouchdb/pouchdb/pull/9010

Reply:
#9010 is passing.  If that gets merged, I think 

```js
    if (rewriteEnabled) {
      // doc.data is what we index, so we need to clone and rewrite it, and clean
      // it up for indexability
      const result = rewrite(theDoc);
      if (result) {
        doc.data = result;
        delete doc.data._attachments;
      } else {
        doc.data = theDoc;
      }   
    } else {
      doc.data = theDoc;
    }
```

can eventually become

```js
    if (rewriteEnabled) {
      // doc.data is what we index, so we need to clone and rewrite it, and clean
      // it up for indexability
      doc.data = rewrite(theDoc);
    } else {
      doc.data = theDoc;
    }

```

Anyway, this PR looks good for now.